fairlearn/datasets/_fetch_boston.py

```diff
@@ -0,0 +1,121 @@
+    PTRATIO  pupil-teacher ratio by town
+    B        1000(Bk - 0.63)^2 where Bk is the proportion of blacks by town
+    LSTAT    % lower status of the population
+    MEDV     Median value of owner-occupied homes in $1000's
+    =======  ======================================================================
+
+
+    Parameters
+    ----------
+    version : integer or 'active', default='active'
+        Version of the dataset. Can only be provided if also ``name`` is given.
+        If 'active' the oldest version that's still active is used. Since
+        there may be more than one active version of a dataset, and those
+        versions may fundamentally be different from one another, setting an
+        exact version is highly recommended.
+
+    cache : boolean, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : optional, default: None
+        Specify another download and cache folder for the datasets. By default
+        all fairlearn data is stored in '~/.fairlearn-data' subfolders.
+
+    as_frame : boolean, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string or categorical). The target is
+        a pandas DataFrame or Series depending on the number of target_columns.
+        The Bunch will contain a ``frame`` attribute with the target and the
+        data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
+        DataFrames or Series as describe above.
+
+    return_X_y : boolean, default=False.
+        If True, returns ``(data.data, data.target)`` instead of a Bunch
+        object.
+
+    warn : boolean, default=True.
+        If True, it raises an extra warning to make users aware of the unfairness
+        aspect of this dataset.
+
+
+    Returns
+    -------
+    dataset : :class:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the following attributes.
+
+        data : ndarray, shape (506, 13)
+            Each row corresponding to the 13 feature values in order.
+            If ``as_frame`` is True, ``data`` is a pandas object.
+        target : numpy array of shape (506,)
+            Each value corresponds to the average
+            house value in units of 100,000.
+            If ``as_frame`` is True, ``target`` is a pandas object.
+        feature_names : list of length 13
+            Array of ordered feature names used in the dataset.
+        DESCR : string
+            Description of the Boston housing dataset.
+
+    (data, target) : tuple if ``return_X_y`` is True
+
+    Notes
+    -----
+    This dataset consists of 506 samples and 13 features. It is notorious for the fairness
+    issues related to the `B` column. There's more information in the references.
+
+    References
+    ----------
+    https://medium.com/@docintangible/racist-data-destruction-113e3eff54a8
+    https://github.com/scikit-learn/scikit-learn/issues/16155
+    """
+    if warn:
```

Comment:
fwiw, I'll put in my final vote against this approach, as previously raised in https://github.com/fairlearn/fairlearn/issues/431#issuecomment-632682693.  I've shared elsewhere (https://github.com/fairlearn/fairlearn/issues/430#issuecomment-632690069, https://github.com/fairlearn/fairlearn/issues/466#issuecomment-640852091, https://github.com/koaning/scikit-fairness/issues/31) a critique of using this dataset in isolation to illustrate the value of algorithmic approaches to assessment and mitigation on a prediction task.  My perception is that I am probably alone in this view :)

I'd love to see more discussion on the use cases for different algorithmic approaches, efforts to connect technical methods to their impact on real world harms, examples and benchmarks set in specific deployment contexts, and to generally see that the work in the project is approaching fairness as a sociotechnical issues.

I've also heard folks on the project talk about the importance of careful language when talking about fairness issues, and that an aspiration for this project is to advance the perspective that fairness is a sociotechnical challenge.  So I'd love to see discussion of the linked articles in that lens, or to see `fairlearn` taking advantage of the opportunity to engage in fairness questions in a deeper way than say `scikit-learn`, which doesn't have the same focus or purpose.

Thanks for reading! 😄 I'll make this my last comment on the topic and step back.  @koaning I hope you and others will consider these perspectives before moving forward in whatever way you all decide is best.

Reply:
@kevinrobinson I've read your comments, but there's a lot of material and various points you're making, so I'm not entirely sure what specifically you're voting against. I suspect it's adding the dataset(?).

It's not entirely clear to me what's objectionable about including the dataset here. From my perspective the presentation (through user guide or examples) is crucial, and that's something @koaning said we should focus on next.
If there's a specific way in which you'd like to present this dataset please feel empowered to propose an approach or submit a PR directly. I think we all value your thorough analysis and I think that could be a great way for anyone interacting with the educational material to learn about this.

In general, I think it's as useful to show failures as it is to show successes (perhaps more!), as long as it's clear that it is a failure and we don't want folks to repeat it.

What do you think? I think this broader discussion will certainly be part of the call on Thursday as well since a lot of us would like to focus on fairness as a sociotechnical issue.

Reply:
@kevinrobinson I also see a lot of material and various points so for me it's also unclear what you're voting against.

The goal here is not to only add one dataset, but many. This is but the beginning. Having a few of these datasets readily available will making it *much* easier to write guides on how to use techniques as well as give context. I'd even go as far as say without examples it'll be *really* hard to explain the merits of using fair techniques. Certainly, there's a sociotechnical context that needs to be adressed (and having datasets means that it's easier to write about those) but without examples of datasets it's going to be really hard to demonstrate to people how (and if) algorithmic fairness techniques work. 

Also note that this PR is merely to add a dataset, there's future PRs which would include blogposts/articles that can give more context. I just think it's much easier to write a blogpost about a dataset if the dataset is readily available.